### PR TITLE
St/withdrawal msg proc

### PR DESCRIPTION
**Pull request by saratnt**
Still todos:
- define a proper amount of gas spent for each operation
- check if in Ethereum there are predefined types of errors that we should use
- log

### CODE REVIEW

File: sdk/src/main/scala/com/horizen/account/state/WithdrawalMsgProcessor.scala [hunk 1]
```diff
@@ -13,14 +12,20 @@ import scorex.util.serialization.{Reader, Writer}
 import scala.collection.JavaConverters.seqAsJavaListConverter
 import scala.util.{Failure, Success, Try}
 
-object WithdrawalMsgProcessor extends AbstractFakeSmartContractMsgProcessor {
+trait WithdrawalRequestProvider extends AbstractFakeSmartContractMsgProcessor {
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/account/state/WithdrawalMsgProcessor.scala:15]: I think WithdrawalRequestProvider should NOT extend AbstractFakeSmartContractMsgProcessor
Instead `object WithdrawalMsgProcessor extends WithdrawalRequestProvider with WithdrawalRequestProvider`

File: sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
-                        val stateDb: StateDB,
-                        messageProcessors: Seq[MessageProcessor]) extends StateView[SidechainTypes#SCAT, AccountStateView] with AccountStateReader {
+class AccountStateView(val metadataStorageView: AccountStateMetadataStorageView,
+                       val stateDb: StateDB,
+                       messageProcessors: Seq[MessageProcessor]) extends StateView[SidechainTypes#SCAT, AccountStateView] with AccountStateReader {
   view: AccountStateView =>
 
   override type NVCT = this.type
 
+  lazy val withdrawalReqProvider: WithdrawalRequestProvider = messageProcessors.find(_.isInstanceOf[WithdrawalRequestProvider]).get.asInstanceOf[WithdrawalRequestProvider]
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala:26]: why not just return `WithdrawalMsgProcessor` object?
Seems safe

**Review by i-Alex — COMMENTED** [review]
Just a few comments.
Then we may merge it to the base branch for the more comprehensive review.

File: sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala (hunk 2 above)
Comment (saratnt) [reply sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala:26]: I wanted to keep the possibility to "inject" the withdrawalmesgprocessor from outside, using the messageProcessors list.
Using directly WithdrawalMsgProcessor object would have removed this possibility.

Comment (i-Alex) [reply sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala:26]: Good. Then no observations.

File: sdk/src/main/scala/com/horizen/account/state/WithdrawalMsgProcessor.scala (hunk 1 above)
Comment (saratnt) [reply sdk/src/main/scala/com/horizen/account/state/WithdrawalMsgProcessor.scala]: Done

**Review by i-Alex — APPROVED** [review]

**Pull request merged**